app/assets/javascripts/components/activity/activity_handler.jsx

```diff
@@ -41,7 +41,7 @@ export const ActivityHandler = createReactClass({
       <div className="activity-handler">
         <SubNavigation links={links}/>
         <Routes>
-          <Route path="recent" element={<RevisionHandler {...this.props}/>} />
+          {this.props.usersLoaded && <Route path="recent" element={<RevisionHandler {...this.props}/>} />}
```

Comment:
Since the revisions are fetched as soon as the `RevisionHandler` component is mounted, the users must be fetched before that happens. I could use a `componentDidUpdate` to run the code only when the users were fetched, but the comment in `app/assets/javascripts/components/revisions/revisions_handler.jsx` says that placing the code here is intentional, so this just felt like an easier solution.

Reply:
okay. I'm not really clear on the logic, but I'm sure it made sense at the time. The user experience is reasonable for when the `recent` route gets loaded directly (it first renders the nav, then renders the Recent Activity table in the 'loading' state once the users are received, I guess).